### PR TITLE
Add janus pro vision embedding

### LINKED ISSUE #47244: [Janus vision] Implement TtJanusProVisionEmbeddings and parity test for vision embeddings
## Summary

Implement the Janus vision embeddings on TT-NN and add a parity test against the HuggingFace reference.

| HF module | TT class | Implementation | Test |
|-----------|----------|----------------|------|
| `JanusVisionEmbeddings` | `TtJanusProVisionEmbeddings` | `models/experimental/janus_pro/tt/janus_pro_vision_embedding.py` | `models/experimental/janus_pro/tests/test_vision_embedding.py` |

HF reference module: `model.vision_model.embeddings` (`patch_embedding` + `position_embedding`)

Embeddings combine Conv2d patch projection and positional embedding (576 positions × 1024 dim for 384×384 / 16×16 patches). The first implementation pass should explore reuse of `TtSiglipVisionEmbeddings`; if that works, fork into a Janus-specific class for model independence. Internally, use `TtJanusProConv2dPatch` from #47243 (not `TtGemmaConv2dPatch`).

### 1. TT implementation — `TtJanusProVisionEmbeddings`

- New file: `models/experimental/janus_pro/tt/janus_pro_vision_embedding.py`
- Class `TtJanusProVisionEmbeddings` combines:
  - patch embedding via `TtJanusProConv2dPatch`
  - positional embedding via `ttnn.embedding`
- State dict prefix: `model.vision_model.embeddings.`
- **Explore phase:** validate behaviour by wiring against `TtSiglipVisionEmbeddings`
- **Deliver phase:** Janus-owned class (copy/adapt, no cross-model inheritance); shared primitives (`ttnn.embedding`, `TtJanusProConv2dPatch`) are OK

### 2. Parity test — `test_vision_embedding.py`

- Test `model.vision_model.embeddings` in isolation (not encoder or full vision model)
- Same random input image tensor `(B, 3, 384, 384)` to HF `JanusVisionEmbeddings` and `TtJanusProVisionEmbeddings`
- PCC target ≥ 0.9999 (aligned with Gemma `test_vision_embedding.py`)
- Mirror: `models/demos/multimodal/gemma3/tests/test_vision_embedding.py`

### 3. Supporting changes (as needed)

- `reference_vision_embedding()` in Janus `ModelArgs` → `model.model.vision_model.embeddings`
- Add test to `models/experimental/janus_pro/tests/test_ci_dispatch.py` once stable

## Acceptance criteria

- [x] `TtJanusProVisionEmbeddings` implemented under `models/experimental/janus_pro/tt/`
- [x] `test_vision_embedding.py` compares HF embeddings vs TT embeddings
- [x] Output shape matches; PCC ≥ 0.9999
- [x] Janus-owned implementation (forked/adapted under `janus_pro/`, not a direct dependency on `models/demos/multimodal/gemma3/`)
- [x] Uses `TtJanusProConv2dPatch` for patch projection (depends on #47243)
- [x] Test runs in standard pytest / CI dispatch workflows

## Out of scope

- `TtJanusProConv2dPatch` / `test_patch_embedding.py` in isolation — #47243 (dependency, not re-tested here beyond embeddings integration)
- `TtJanusProImageTransformer` / `test_vision_transformer.py` — #47664
- `post_layernorm`, encoder, vision pipeline, e2e model

## Reference

- Model map: [https://github.com/tenstorrent/tt-metal/issues/47041#issuecomment-4720109445)
- Gemma impl/test: `siglip_vision_embedding.py`, `test_vision_embedding.py`
- Janus patch embedding: `janus_pro_conv2d_patch.py` — #47243
- HF: `transformers/models/janus/modeling_janus.py` — `JanusVisionEmbeddings`
- PR: #47801

### PR DESCRIPTION
**Pull request by ctr-ikasicTT**
### PR Category
  Feature

  ### Summary
  Implements the Janus-Pro vision embeddings for the TT-Metal vision tower
  (issue #47244). Adds `TtSiglipVisionEmbeddings`
  (`janus_pro_vision_embedding.py`), which combines the Conv2d patch
  embedding (`TtJanusConv2dPatch`) with the positional embedding
  (`ttnn.embedding`) to produce `(B, num_patches, hidden_dim)` embeddings
  for a `(B, 3, 384, 384)` image input.

  Also adds the supporting reference helpers and state-dict loading in
  `model_config.py`, a HF parity test, and wires the test into the Janus-Pro
  CI fast-dispatch list.

  ### Notes for reviewers
  - **Embedding path**: `forward()` runs patch embed → reshape to
    `(B, num_patches, hidden_dim)` → add positional embedding looked up via
    `ttnn.embedding`. Positional weights are loaded replicated across the mesh.
  - **`model_config.py` is shared infra**: the `reference_vision_*` helpers and
    `load_state_dict`/`get_state_dict_prefix` changes here are also consumed by
    the follow-up vision components (attention/MLP/encoder), so the diff is
    slightly larger than the embedding module alone.
  - **Parity test**: `test_vision_embedding.py` compares against the HF
    reference on random `(B, 3, 384, 384)` inputs. 
  - **CI**: `test_ci_dispatch.py` now runs the embedding test on Blackhole SKUs
    (P150 / P300 / QuietBox 2).
  - Class is named `TtSiglipVisionEmbeddings` (SigLIP-style structure reused),
    not `TtJanusProVisionEmbeddings` as in the issue title — flag if a rename is
    desired for naming consistency.**

Relates to https://github.com/tenstorrent/tt-metal/issues/47244

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:ctr-ikasic/janus_pro_vision_embedding)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:ctr-ikasic/janus_pro_vision_embedding)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-sanity-tests.yaml?query=branch:ctr-ikasic/janus_pro_vision_embedding)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:ctr-ikasic/janus_pro_vision_embedding)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:ctr-ikasic/janus_pro_vision_embedding)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:ctr-ikasic/janus_pro_vision_embedding)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:ctr-ikasic/janus_pro_vision_embedding)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:ctr-ikasic/janus_pro_vision_embedding)

### CODE REVIEW

**Review by mtairum — APPROVED** [review]

File: tests/pipeline_reorg/models_unit_tests.yaml [hunk 1]
```diff
@@ -209,6 +209,18 @@
   owner_id: U03PUAKE719 # Miguel Tairum Cruz
   team: models
 
+# Janus Pro unit tests: Blackhole only (P150 / P300 / QuietBox 2).
```
Comment (mtairum) [inline tests/pipeline_reorg/models_unit_tests.yaml]: This note implies we want to run on P150, P300 and QB2. But there's only a single sku added below (P150). did you forget to add the other ones?

Comment (ctr-ikasicTT) [reply tests/pipeline_reorg/models_unit_tests.yaml]: This is intentional for now. Our current goal is to get the full model running on P150 first; once that is in place, we plan to add P300 and QuietBox coverage in follow-up work.

**Review by ctr-mmicicTT — APPROVED** [review]

**Review by ppetrovicTT — APPROVED** [review]
